### ISSUE TITLE
use hash in filename?

### ISSUE THREAD

**Issue by Ginhing** [SimenB/add-asset-html-webpack-plugin#56]
Thank you for your great job! 

Could I use the hash in the output file name just like `mydll.[hash].js` ?

**Comment by SimenB** [comment]
>Could I use the hash in the output file name just like `mydll.[hash].js`?

What do you mean? Do you have some example config of what you want?

>Thank you for your great job!

You're welcome, hope it's useful!

**Comment by Ginhing** [comment]
In current version there is a 'hash' option in it right? I set it true and
built, the file I defined in 'filepath' was copied into the output path and
was referenced in the html generated with a hash as a query string such as
`mydll.js?xxxxxxxxxxx`.

Is there an option make the hash to be inserted into the filename instead
of just query string?

On Sun, Apr 9, 2017, 02:16 Simen Bekkhus <notifications@github.com> wrote:

Could I use the hash in the output file name just like mydll.[hash].js?

What do you mean? Do you have some example config of what you want?

Thank you for your great job!

You're welcome, hope it's useful!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/SimenB/add-asset-html-webpack-plugin/issues/56#issuecomment-292736013>,
or mute the thread

**Comment by SimenB** [comment]
Ah, ok. That does not exist at the moment no. If you can figure out some API that's not too clunky, I'd be happy to merge a PR for it, though 😄

**Comment by Ginhing** [comment]
It seems that [this method](https://github.com/SimenB/add-asset-html-webpack-plugin/blob/e0ebc4c/src/addAllAssetsToCompilation.js#L52) of `html-webpack-plugin` has a limitation on producing a hash.

**Comment by SimenB** [comment]
Yes, that's a good point. We'd have to either do some work in html-webpack-plugin, or doe something smart in this method: https://github.com/SimenB/add-asset-html-webpack-plugin/blob/e0ebc4c/src/addAllAssetsToCompilation.js#L24-L29

**Comment by nikolaas** [comment]
Hello. Is there any progress in the task? It will be useful for me.

**Comment by nikolaas** [comment]
In my task, I need to use a hash of the compilation in the file name, so now I have to use `html-webpack-include-assets-plugin` and `copy-webpack-plugin`. But I saw that you want to merge your project with  `html-webpack-include-assets-plugin`. I would be glad to use `add-asset-html-webpack-plugin` if it allowed to specify the hash of the compilation in the asset name.

**Comment by artbase** [comment]
Hi,
Do you do something with the task? It will be useful for me, too.

**Comment by SimenB** [comment]
I'd love to see a PR supporting this 🙂

**Comment by yuyichen** [comment]
` `` 
output: {
    filename: '[name].dll-[hash:7].js',
    path: path.resolve(__dirname, '../dist/static/vendor/'),
    library: '[name]_[hash]',
  },
```

**Comment by wuliupo** [comment]
if there is a CDN pointing to source host, abcd.css?t=1234 will not be updated to new content. 
So we need abcd.1234.css
